tests/functional/test_search.py

```diff
@@ -106,32 +105,6 @@ def test_search_missing_argument(script):
     assert 'ERROR: Missing required argument (search query).' in result.stderr
 
 
-@pytest.mark.network
```

Comment:
I would keep these tests here since they are more about testing the search command as opposed to the XMLRPC client specifically. It doesn't look like there are any direct tests of `PipXmlrpcTransport` so we just need to move the class, no tests.

Reply:
I felt like hitting `SearchCommand.run` would test the `PipXmlrpcTransport` class since it calls it. But sure, let me move it

Reply:
Thank you! To hopefully provide more insight:

1. we differentiate "functional" tests from "unit" tests, where the former will usually invoke a pip command and the latter will test a class directly. These should each be in their respective folder in `tests`.
2. we'll typically organize the tests based on either topic or the highest-level or  "outermost" interface we're using in the test itself. In this case `pip search`. You can imagine a similar argument for keeping all of the `tests/functional/test_install*` tests in place. While they use a lot of pip internals, they are accessed only through invocations of `pip install`.

1 is a more important consideration than 2. 2 is of at least medium importance.